src/box/index.cc

```diff
@@ -1010,6 +1014,56 @@ generic_index_create_iterator_with_offset(struct index *base,
 	return NULL;
 }
 
+#if defined(ENABLE_READ_VIEW)
+
+int
+generic_index_read_view_create_iterator_with_offset(
+	struct index_read_view *rv, enum iterator_type type,
+	const char *key, uint32_t part_count,
+	const char *pos, uint32_t offset,
+	struct index_read_view_iterator *it)
+{
+	if (index_read_view_create_iterator_after(rv, type, key, part_count,
+						  pos, it) != 0)
+		return -1;
+
+	/* Skip the required amount of tuples. */
+	for (size_t i = 0; i < offset; i++) {
+		/* The fiber slice is checked in the next function. */
+		struct tuple *skipped_tuple;
+		if (box_index_read_view_iterator_next(it, &skipped_tuple) != 0)
```

Comment:
Since you skip tuples anyway, you can use `next_raw` here. Then you wouldn't need to include `index_read_view.h`, nor define the function under `ENABLE_READ_VIEW`.

Reply:
Right, used the raw iterator step here and in the `count()` generic implementation. Thanks.